jellyfish/api/_room_api.py

```diff
@@ -119,17 +121,22 @@ def delete_peer(self, room_id: str, peer_id: str) -> None:
         return self._request(room_delete_peer, id=peer_id, room_id=room_id)
 
     def add_component(
-        self, room_id: str, options: Union[ComponentOptionsHLS, ComponentOptionsRTSP]
-    ) -> Union[ComponentHLS, ComponentRTSP]:
+        self,
+        room_id: str,
+        options: Union[ComponentOptionsFile, ComponentOptionsHLS, ComponentOptionsRTSP],
+    ) -> Union[ComponentFile, ComponentHLS, ComponentRTSP]:
         """Creates component in the room"""
 
-        if isinstance(options, ComponentOptionsHLS):
+        if isinstance(options, ComponentOptionsFile):
+            component_type = "file"
+        elif isinstance(options, ComponentOptionsHLS):
             component_type = "hls"
         elif isinstance(options, ComponentOptionsRTSP):
             component_type = "rtsp"
         else:
             raise ValueError(
-                "options must be either ComponentOptionsHLS or ComponentOptionsRTSP"
+                "options must be ComponentFile, ComponentOptionsHLS"
+                "or ComponentOptionsRTSP"
             )
```

Comment:
Could we use `match ... case` here? Similarly like in this [issue](https://stackoverflow.com/a/67524642)

Reply:
This is python 3.10 feature, so we can't use it if we want to support older python versions.